### ISSUE TITLE
Agda executable cannot be found

### ISSUE THREAD

**Issue by bixuanzju** [syl20bnr/spacemacs#6644]
#### Description :octocat:
Agda executable cannot be found when first launching Emacs

#### Reproduction guide :beetle:
- Add `agda`  to your `~/.spacemacs`.
- Start Emacs
- Look at the *Messages* buffer

*Observed behaviour:* :eyes: :broken_heart:
>(Spacemacs) Warning: Agda not detected, be sure that Agda binaries are available in your PATH or check the installation instructions in the README file.

**Strangely, if I then restart Emacs by `SPC q R`, then the waning is gone.**

*Expected behaviour:* :heart: :smile:
No such warning

#### Possible cause

I suspect this is due to the strange loading order of layers. Specifically, in OSX, we need the package `exec-path-from-shell` to properly set `exec-path`. However, by looking at the *Messages* buffer (which I attach below), I found the agda layer is loaded before loading spacemacs-base, in which `exec-path-from-shell` is initialized, thus Emacs cannot find `agda-mode` executable.

Actually some of the configuration layers are loaded **before** the base layer, which could potentially cause problems.

```
...
Loading /Users/jeremybi/.emacs.d/layers/+lang/agda/packages.el (source)...done
Loading /Users/jeremybi/dotfile/emacs/mylayers/proof-general/packages.el (source)...done
Loading /Users/jeremybi/.emacs.d/layers/+distributions/spacemacs/layers.el (source)...
Loading /Users/jeremybi/.emacs.d/layers/+distributions/spacemacs-base/packages.el (source)...done
...
```

#### System Info :computer:
- OS: darwin
- Emacs: 25.0.95.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. 9773ad6)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm
 (auto-completion :variables auto-completion-enable-sort-by-usage t)
 emacs-lisp
 (git :variables git-magit-status-fullscreen t)
 markdown org
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 spell-checking
 (syntax-checking :variables syntax-checking-enable-tooltips nil)
 (version-control :variables version-control-diff-tool 'diff-hl)
 osx idris
 (haskell :variables haskell-completion-backend 'intero haskell-enable-hindent-style "johan-tibell")
 (latex :variables latex-build-command "LatexMk")
 (evil-snipe :variables evil-snipe-enable-alternate-f-and-t-behaviors t)
 ocaml lua sml yaml
 (c-c++ :variables c-c++-enable-clang-support t)
 html agda proof-general)
```

**Comment by TheBB** [comment]
> I suspect this is due to the strange loading order of layers.

No, the packages.el files can be loaded in any order because they just define functions, nothing else. The interesting thing is in which order the packages are configured, namely alphabetically, which puts agda before exec-path-from-shell.

We can override this order with `:step`, so check if https://github.com/syl20bnr/spacemacs/pull/6646 solves the problem for you.

**Comment by bixuanzju** [comment]
@TheBB Yes, works like a charm!

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
